flume-ng-sources/flume-jms-source/src/main/java/org/apache/flume/source/jms/JMSSource.java

```diff
@@ -121,6 +121,13 @@ protected void doConfigure(Context context) throws FlumeException {
     pollTimeout = context.getLong(JMSSourceConfiguration.POLL_TIMEOUT,
         JMSSourceConfiguration.POLL_TIMEOUT_DEFAULT);
 
+    clientId = Optional.fromNullable(context.getString(JMSSourceConfiguration.CLIENT_ID));
+
+    createDurableSubscription = context.getBoolean(
+        JMSSourceConfiguration.CREATE_DURABLE_SUBSCRIPTION, false);
```

Comment:
Would you mind using constants instead of magic value? ie DEFAULT_CREATE_DURABLE_SUBSCRIPTION and DEFAULT_DURABLE_SUBSCRIPTION_NAME